ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/search_tree/algorithms/SearchTreeBloomer.java

```diff
@@ -146,13 +162,32 @@ List<NetworkActionCombination> removeCombinationsWhichExceedMaxNumberOfRa(List<N
         return filteredNaCombinations;
     }
 
-    List<NetworkActionCombination> removeCombinationsWhichExceedMaxNumberOfRaPerTso(List<NetworkActionCombination> naCombinations, Leaf fromLeaf) {
+    Map<NetworkActionCombination, Boolean> removeCombinationsWhichExceedMaxNumberOfRaPerTso(Map<NetworkActionCombination, Boolean> naCombinations, Leaf fromLeaf) {
```

Comment:
could you try to refactor to do this more efficiently ?

Reply:
We did not find a more efficient way to do this. If you want we can extract a method from this one to make it lighter.